src/core/nodeOps.ts

```diff
@@ -39,7 +39,8 @@ export const nodeOps: RendererOptions<TresObject, TresObject> = {
       if (props?.object === undefined) logError('Tres primitives need a prop \'object\'')
       const object = props.object as TresObject
```

Comment:
Hey @andretchen0, thanks for checking it, I ended up modifying this part because we need to clone the `prop.object`  to avoid overwriting the original three object when swapping happens.